ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ImmDB.hs

```diff
@@ -350,22 +350,17 @@ appendBlock db@ImmDB{..} b = withDB db $ \imm -> case isEBB b of
 -- | Wrapper around 'ImmDB.streamBinaryBlobs' that 'allocate's the iterator in
 -- the 'ResourceRegistry' so that 'ImmDB.iteratorClose' is registered as the
 -- clean-up action.
---
--- When the returned iterator is closed, it will be 'release'd from the
--- 'ResourceRegistry'.
 registeredStream :: (MonadMask m, MonadSTM m, MonadFork m, HasHeader blk)
                  => ImmDB m blk
                  -> ResourceRegistry m
                  -> Maybe (SlotNo, HeaderHash blk)
                  -> Maybe (SlotNo, HeaderHash blk)
                  -> m (ImmDB.Iterator (HeaderHash blk) m Lazy.ByteString)
 registeredStream db registry start end = do
-    (key, it) <- allocate registry
+    (_key, it) <- allocate registry
       (\_key -> withDB db $ \imm -> ImmDB.streamBinaryBlobs imm start end)
       (iteratorClose db)
     return it
-      { ImmDB.iteratorClose = release registry key
```

Comment:
I see. Depending on what the final `ResourceRegistry` will look like, we might have to revert this.

Potential problematic scenario: if this registry is only closed when the node is shut down, then not releasing a pre-emptively closed iterator means that the registry keeps growing.

Reply:
This is the cause of the latest CI failure in `test-storage`, so I'm hoping PR 929 lets me discard this part of the diff.